### PR TITLE
`Quiz exercises`: Fix batch quiz editing when batch is added

### PR DESCRIPTION
**Pull request by rriyaldhi**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

An instructor can still visit the quiz edit page even though the batched quiz has batches already by clicking the quiz id on the exercises page or by visiting directly the edit page url.

### Description

- Disable the quiz id link to the edit page when the batched quiz has batches.
- Show an unauthorized popup message and disable the page if the quiz is not editable and the edit page link is visited.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course

1. Log in to Artemis
2. Navigate to Course Administration
3. Select the Course
4. Navigate to Exercises
5. Create a new Quiz Exercise
6. Set Batch Mode to **Batched**
7. Add 1 Question of any type
8. Click Save
9. Click Back
10. Click Edit on the newly created quiz
11. Verify that the quiz is still editable
12. Click Back
13. Click **+ Add Batch**
14. Verify that the edit button is disabled and the quiz id is not clickable
15. Verify that when the edit link is visited, the page shows an unauthorized popup message and the page does not show anything.
16. Verify that importing the quiz exercise works.

Note: See the screencast for the detailed steps for testing.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| quiz-exercise-detail.component.ts | 79.79% | 94.66% |
| quiz-exercise.component.ts | 58.1% | 88.52% |
| quiz-manage-util.service.ts | 100% | 100% |

### Screenshots

https://user-images.githubusercontent.com/10885503/210074261-88cb4a38-3afd-45b7-9060-7ce7c24b8cf9.mov

### CODE REVIEW

**Review by tunargs — DISMISSED** [review]
Tested on ts1. works exactly as in the screencast

**Review by minrows — APPROVED** [review]
Tested on ts2. Works as explained in the description. LGTM 👍

**Review by Santia-go — APPROVED** [review]
Tested on TS0
The bug has been fixed. 
I tried editing the link and also the address schema. It is not possible to edit. 
image

File: src/main/webapp/app/exercises/quiz/shared/quiz-manage-util.service.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import { QuizExercise, QuizMode, QuizStatus } from 'app/entities/quiz/quiz-exercise.model';
+
+@Injectable({ providedIn: 'root' })
+export class QuizManageUtil {
+    /**
+     * Check if quiz is editable
+     * @param quizExercise the quiz exercise which will be checked
+     * @return {boolean} true if the quiz is editable and false otherwise
+     */
+    isQuizEditable(quizExercise: QuizExercise): boolean {
```
Comment (krusche) [inline src/main/webapp/app/exercises/quiz/shared/quiz-manage-util.service.ts:11]: Is it really necessary to create such a class with root injection? The method could also be simply static and be imported from a util typescript file

**Review by krusche — CHANGES_REQUESTED** [review]
See my inline comment

**Review by pellumbbaboci — APPROVED** [review]
Tested on TS1, works as expected :+1:

File: src/main/webapp/app/exercises/quiz/shared/quiz-manage-util.service.ts (hunk 1 above)
Comment (rriyaldhi) [reply src/main/webapp/app/exercises/quiz/shared/quiz-manage-util.service.ts]: I followed ShortAnswerUtil class. Thanks for the suggestion! I changed this to an exported function.

**Review by Santia-go — APPROVED** [review]
Re-tested on TS2 
The behavior is correct. I ti not possible to edit the quiz once the batch is added. Even changing the link or clicking in the navigator part. 
Re-approved 
image

**Review by pellumbbaboci — APPROVED** [review]
Tested on ts2, works as expected. Re-approved

**Review by minrows — APPROVED** [review]
Re-tested on ts0, works as expected. Reapproved. 👍

**Pull request merged**